ompy/error_finder.py

```diff
@@ -0,0 +1,433 @@
+
+        Warnings:
+            Will raise a warning if there are members in the ensemble that
+            contains one or more nan's. This is mostly to inform the user
+            and shouldn't be an issue later on.
+
+        TODO:
+            - Mitigation when cut_nan() results in different length vectors
+              of different members of the ensemble. (e.g. len(nld[0]) = 10,
+              len(nld[1]) = 9).
+
+        """
+
+        # Ensure that the length of the NLDs and GSFs are equal
+        assert len(_nlds) == len(_gsfs), \
+            "Length of nlds and gsfs doesn't match"
+
+        N = len(_nlds)
+
+        self.LOG.debug("Processing an ensemble with %d members", N)
+
+        # Make copy to ensure that we don't overwrite anything
+        nlds = [nld.copy() for nld in _nlds]
+        gsfs = [gsf.copy() for gsf in _gsfs]
+
+        self.LOG.debug("Before removing nan: %d NLD values and %d GSF values",
+                       len(nlds[0]), len(gsfs[0]))
+
+        # Ensure that we have in MeV and that there isn't any nan's.
+        is_nan = False
+        for nld, gsf in zip(nlds, gsfs):
+            nld.to_MeV()
+            gsf.to_MeV()
+            is_nan = np.isnan(nld.values).any() or np.isnan(gsf.values).any()
+            nld.cut_nan()
+            gsf.cut_nan()
+        if is_nan:
+            self.LOG.warning(f"NLDs and/or γSFs contains nan's."
+                             " They will be removed")
+
+        # Next we will ensure that all members have all energies
+        if (not all_equal([len(nld) for nld in nlds]) or
+           not all_equal([len(gsf) for gsf in gsfs])):
+            self.LOG.warning("Some members of the ensemble have different "
+                             "lengths. "
+                             "Consider re-binning or changing limits.")
+            nlds = self.keep_only(nlds)
+            gsfs = self.keep_only(gsfs)
+
+        self.LOG.debug("After removing nan: %d NLD values and %d GSF values",
+                       len(nlds[0]), len(gsfs[0]))
+
+        # Next we can extract the important bits
+        E_nld = nlds[0].E.copy()
+        E_gsf = gsfs[0].E.copy()
+
+        M_nld = len(E_nld)
+        M_gsf = len(E_gsf)
+
+        nld_obs = []
+        gsf_obs = []
+
+        idx_nld = np.tile(np.arange(M_nld, dtype=int), (N-1, 1))
+        idx_gsf = np.tile(np.arange(M_gsf, dtype=int), (N-1, 1))
+
+        for n, (nld, gsf) in enumerate(zip(nlds, gsfs)):
+
+            # Make a copy of the values arrays
+            nld_array = [nld.values.copy() for nld in nlds]
+            gsf_array = [gsf.values.copy() for gsf in gsfs]
+
+            del nld_array[n]
+            del gsf_array[n]
+
+            nld_array = np.array(nld_array)
+            gsf_array = np.array(gsf_array)
+
+            # Set the observed data
+            nld_obs.append(nld.values[idx_nld]/nld_array)
+            gsf_obs.append(gsf.values[idx_gsf]/gsf_array)
```

Comment:
- This is somewhat funky. Why do you need to manually delete the array -- but after setting it by a copy?
I if I understand this correctly, it is because you want to get rid of the reference dataset. Maybe some comments would be appreciated here.